servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/LoggingHttpLifecycleObserver.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.utils;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.http.api.HttpHeaders;
+import io.servicetalk.http.api.HttpLifecycleObserver;
+import io.servicetalk.http.api.HttpRequestMetaData;
+import io.servicetalk.http.api.HttpResponseMetaData;
+import io.servicetalk.logging.api.LogLevel;
+import io.servicetalk.logging.slf4j.internal.FixedLevelLogger;
+import io.servicetalk.transport.api.ConnectionInfo;
+
+import javax.annotation.Nullable;
+
+import static io.servicetalk.logging.slf4j.internal.Slf4jFixedLevelLoggers.newLogger;
+import static io.servicetalk.utils.internal.ThrowableUtils.combine;
+import static java.lang.System.nanoTime;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/**
+ * Logging implementation of {@link HttpLifecycleObserver}.
+ */
+public final class LoggingHttpLifecycleObserver implements HttpLifecycleObserver {
+
+    private final FixedLevelLogger logger;
+
+    /**
+     * Create a new instance.
+     *
+     * @param loggerName The name of the logger to use.
+     * @param logLevel The level to log at.
+     */
+    public LoggingHttpLifecycleObserver(final String loggerName, final LogLevel logLevel) {
+        this.logger = newLogger(loggerName, logLevel);
```

Comment:
@bondolo made a valid comment earlier that users may expect errors to be logged at `ERROR` level. Should we consider taking only the logger name and enforce level internally?
- INFO - if both complete successfully
- WARN - if request or response was canceled
- ERROR - if request or response failed

We can also have two ctor overloads: one with a fixed logger level, another one as described above. WDYT?

Reply:
I would reduce the level for the happy path to `DEBUG`, but otherwise agree. For some users who use cancel as part of their normal use `WARN` might be too severe or noisy, perhaps `INFO` or `DEBUG`. If they are doing a "search auto completion" service and cancel each time a character is typed then it would certainly be too noisy at WARN level.

Reply:
Thanks for the input!
It's hard to decide which logger level will be perfect for users without hearing any feedback from them. Let's start with a fixed level and we can add more ctor overloads (or a builder) when we learn more from users.